packages/daimo-common/src/daimoLink.ts

```diff
@@ -301,3 +314,66 @@ function parseDaimoLinkInner(link: string): DaimoLink | null {
       return null;
   }
 }
+
+function parseOldDaimoLinkRequest(parts: string[]): DaimoLinkRequest | null {
+  if (parts.length !== 4) return null;
+  const recipient = parts[1];
+  const dollarNum = parseFloat(zDollarStr.parse(parts[2]));
+  if (!(dollarNum > 0)) return null;
+  const dollars = dollarNum.toFixed(2) as DollarStr;
+  const requestId = `${BigInt(parts[3])}` as BigIntStr;
+
+  if (dollars === "0.00") return null;
+
+  const toChain = base;
+  const toCoin = baseUSDC;
+
+  return { type: "request", requestId, recipient, dollars, toCoin, toChain };
+}
+
+function parseNewDaimoLinkRequest(url: URL): DaimoLinkRequest | null {
+  const id = url.searchParams.get("id");
+  const to = url.searchParams.get("to");
+  const n = url.searchParams.get("n");
+  const c = url.searchParams.get("c");
+  const t = url.searchParams.get("t");
+
+  if (!to || !c || !t) return null;
```

Comment:
to avoid cross-chain request links to Daimo accounts, we could check here that `to` is not an account name. eg

```
if (isValidAccountName(to)) return null;
```